x-pack/elastic-agent/pkg/core/monitoring/beats/beats_monitor.go

```diff
@@ -122,6 +122,7 @@ func (b *Monitor) EnrichArgs(spec program.Spec, pipelineID string, args []string
 		appendix = append(appendix,
 			"-E", "http.enabled=true",
 			"-E", "http.host="+endpoint,
+			"-E", "http.pprof.enabled=true",
```

Comment:
If monitoring is enabled we always enable also pprof?

Reply:
I wonder if this might be unexpected from a user perspective. pprof exposes quite a bit more information than monitoring as it exposes some of the internals of the process. At the same time, whoever gets the monitoring data likely already knows about the internals.

@simitt @scunningham Any concerns with this?

Reply:
Can you help me understand how the current config options play together: 
```
agent.monitoring:
  enabled: false
  logs: false
  metrics: false
  http:
    enabled: true 
```

Will metrics be exposed via the `http` endpoint?

Reply:
We should not enable pprof unless customer explicitly turns it on.  The HTTP interface is not defended and an attacker would be able to potentially steal secrets from the heap dump.

Reply:
`http.pprof.enabled` is a libbeat setting that we are passing to the beats to expose the pprof options over the http endpoint.
When the agent passes it above, the http interface is bound to a local unix socket (or windows npipe)

@scunningham, when the elastic-agent starts a beat, the beat will bind the interface to a socket even if `agent.monitoring.enabled: false`, is your request to disable automatically enabling the pprof endpoints for these beats unless explicitly enabled in this case as well?

Reply:
Are you saying we automatically enable pprof even if not explicitly enabled?  That's not great.